server/test/test_nan_scanpy_engine.py

```diff
@@ -16,69 +19,52 @@ def setUp(self):
             "obs_names": None,
             "var_names": None,
             "diffexp_lfc_cutoff": 0.01,
-            "nan_to_num": False,
         }
         with warnings.catch_warnings():
             warnings.simplefilter("ignore", category=UserWarning)
             self.data = ScanpyEngine("server/test/test_datasets/nan.h5ad", self.args)
             self.data._create_schema()
-        self.args_nan = dict(self.args)
-        self.args_nan["nan_to_num"] = True
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore", category=UserWarning)
-            self.data_nan = ScanpyEngine(
-                "server/test/test_datasets/nan.h5ad", self.args_nan
-            )
-            self.data_nan._create_schema()
 
     def test_load(self):
         with self.assertWarns(UserWarning):
-            ScanpyEngine("server/test/test_datasets/nan.h5ad", self.args_nan)
+            ScanpyEngine("server/test/test_datasets/nan.h5ad", self.args)
 
     def test_init(self):
         self.assertEqual(self.data.cell_count, 100)
         self.assertEqual(self.data.gene_count, 100)
         epsilon = 0.000_005
         self.assertTrue(self.data.data.X[0, 0] - -0.171_469_51 < epsilon)
 
-        self.assertEqual(self.data_nan.cell_count, 100)
-        self.assertEqual(self.data_nan.gene_count, 100)
-        epsilon = 0.000_005
-        self.assertTrue(self.data_nan.data.X[0, 0] - -0.171_469_51 < epsilon)
-
     def test_dataframe(self):
-        data_frame_obs = json.loads(self.data_nan.data_frame(None, "obs"))
-        self.assertEqual(len(data_frame_obs["var"]), 100)
-        self.assertEqual(len(data_frame_obs["obs"]), 100)
-        data_frame_var = json.loads(self.data_nan.data_frame(None, "var"))
-        self.assertEqual(len(data_frame_var["var"]), 100)
-        self.assertEqual(len(data_frame_var["obs"]), 100)
+        data_frame_var = decode_fbs.decode_matrix_FBS(self.data.data_frame_to_fbs_matrix(None, "var"))
+        self.assertIsNotNone(data_frame_var)
+        self.assertEqual(data_frame_var["n_rows"], 100)
+        self.assertEqual(data_frame_var["n_cols"], 100)
+        self.assertTrue(math.isnan(data_frame_var["columns"][3][3]))
+
+        with self.assertRaises(ValueError) as cm:
+            data_frame_obs = decode_fbs.decode_matrix_FBS(self.data.data_frame_to_fbs_matrix(None, "obs"))
```

Comment:
causing failing flake8. Just need to remove `data_frame_obs = ` assignment since the variable isn't used again.

Reply:
fixed

Reply:
note - the same error occurred in all of the exception tests, but wasn't noted by flake8 due to a reused variable.  I'll fix it everywhere.